ios-training/Controller/ViewController.swift

```diff
@@ -35,48 +36,48 @@ class ViewController: UIViewController {
 
     @IBAction func reloadButton(_ sender: UIButton) {
         activityIndicator.startAnimating()
-        weatherModel.fetchWeather()
+        weatherModel.fetchWeather() { result in
+            DispatchQueue.main.async {
+                self.activityIndicator.stopAnimating()
+                self.receiveWeather(result: result)
+            }
+        }
     }
     
     @IBAction func closeButton(_ sender: UIButton) {
         dismiss(animated: true)
     }
     
-}
-
-extension ViewController: WeatherDelegate {
-    func loadWeather(_ result: Result<Response, YumemiWeatherError>) {
-        DispatchQueue.main.async {
-            self.activityIndicator.stopAnimating()
-            switch result {
-            case .success(let response):
-                self.weatherImageView.image = UIImage(named: response.weatherCondition.rawValue)
-                
-                switch response.weatherCondition {
-                case .sunny:
-                    self.weatherImageView.tintColor = .red
-                case .cloudy:
-                    self.weatherImageView.tintColor = .gray
-                case .rainy:
-                    self.weatherImageView.tintColor = .blue
-                }
-                
-                self.minTemperature.text = String(response.minTemperature)
-                self.maxTemperature.text = String(response.maxTemperature)
-                
-            case .failure(let error):
-                let errorMessage: String
-                switch error {
-                case .invalidParameterError:
-                    errorMessage = "invalidParameterError"
-                case .unknownError:
-                    errorMessage = "unknownError"
-                }
-                
-                let alert = UIAlertController(title: "Error", message: errorMessage, preferredStyle: .alert)
-                alert.addAction(UIAlertAction(title: "OK", style: .default))
-                self.present(alert, animated: true)
+    func receiveWeather(result: Result<Response, YumemiWeatherError>) {
```

Comment:
viewController内部から飲み参照されていると思いますので、privateにするのが望ましいです。
```suggestion
    private func receiveWeather(result: Result<Response, YumemiWeatherError>) {
```